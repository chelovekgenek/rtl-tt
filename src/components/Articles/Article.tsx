import React, { useMemo, useCallback } from "react"
import { IStateProps } from "./Articles.container"

import * as Styles from "./Article.styled"
import { getCropBy, truncateText } from "helpers"

interface IProps {
  data: IStateProps["data"][number]
}

export const Article: React.FC<IProps> = ({ data }) => {
  const image = useMemo(() => getCropBy(data.afbeelding.crops, "type", "square_small"), [data.afbeelding.crops])
  const description = useMemo(() => truncateText(data.lead, 240), [data.lead])
  const handleRedirect = useCallback(() => window.location.assign(data.urlAlias), [data.urlAlias])
  return (
    <Styles.Container onClick={handleRedirect}>
      {Boolean(image) && (
        <Styles.Section>
          <Styles.ImageWrapper>
            <Styles.Image src={image!.path} alt={data.id} />
          </Styles.ImageWrapper>
        </Styles.Section>
      )}
      <Styles.Section>
        <Styles.Label>{data.labelValue}</Styles.Label>
        <Styles.Title>{data.titel}</Styles.Title>
        <p>{description}</p>
      </Styles.Section>
    </Styles.Container>
  )
}
