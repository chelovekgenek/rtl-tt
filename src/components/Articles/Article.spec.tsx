import React from "react"
import { shallow } from "enzyme"

import bundleItems from "mocks/bundle-items.json"

import { Article } from "./Article"
import { Container } from "./Article.styled"

describe("Article", () => {
  const render = (props: Partial<React.ComponentProps<typeof Article>> = {}) =>
    shallow(<Article data={bundleItems[0]} {...props} />)

  const { location } = window
  beforeAll(() => {
    delete window.location
    window.location = { assign: jest.fn() }
  })

  afterAll(() => {
    window.location = location
  })

  it("should render", () => {
    expect(render()).toMatchSnapshot()
  })
  it("should redirect when Container is clicked", () => {
    const fn = jest.spyOn(window.location, "assign")
    render()
      .find(Container)
      .simulate("click")
    expect(fn).toHaveBeenCalledWith(bundleItems[0].urlAlias)
  })
})
