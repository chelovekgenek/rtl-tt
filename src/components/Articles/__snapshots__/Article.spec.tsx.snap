// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should render 1`] = `
<styled.li
  onClick={[Function]}
>
  <styled.div>
    <styled.div>
      <styled.img
        alt={4245471}
        src="https://www.rtlboulevard.nl/sites/default/files/styles/square_small/public/content/images/2018/06/19/ANP-56919532.jpg?itok=lTlIv6qo"
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.h3>
      De uitschieters van 2018
    </styled.h3>
    <styled.h2>
      Trijntje Oosterhuis onherkenbaar veel afgevallen
    </styled.h2>
    <p>
      Onze geliefde souldiva Trijntje Oosterhuis is aanzienlijk veel kilo's verloren! De zangeres plaatst foto's op haar Instagram waar te zien is dat ze een wespentaille heeft waar de meeste vrouwen stront jaloers op zullen zijn. 
    </p>
  </styled.div>
</styled.li>
`;
